statesman-engine/src/main/java/io/appform/statesman/engine/ProviderSelector.java

```diff
@@ -0,0 +1,10 @@
+package io.appform.statesman.engine;
+
+import io.appform.statesman.model.Workflow;
+
+import java.util.Set;
+
+public interface ProviderSelector {
+
+    String provider(String providerType, Set<String> configuredProviders, Workflow workflow);
```

Comment:
Return the full provider object, you'll need it somewhere at a later point of time. Also what happens if nothing is found? This should return an optional to force the client to think about that scenario

Reply:
This should _only_ take the `providerType`. It does not need workflow. It should find the list of providers from the provider store based on the` providerType`. should be renamed to `usecase`

Reply:
I see changes missing from stored provider and store... that will be needed to stitch the whole thing end to end.